### DISCUSSION TITLE
Is it possible to keep some inline suggestions?

### DISCUSSION THREAD

**Discussion by adrennhoff** [posit-dev/positron#4642] [Q&A]
I am really enjoying my initial exploration with Positron but I was hoping to eliminate/reduce one thing. 

Suppose I am creating a simple pandas dataframe using a dictionary (defining all in one line). I could do something like:

df = pd.DataFrame({
       'Name' : ['Bob', 'Sally'],
       'Age' : [25, 30]
})

With the inline suggestions, once I type "pd." it shows me options, including 'DataFrame'. I like this -- it's helpful.

But, once I type the quotes where I want to define my first column, the inline suggestion shows me two things: it offers a dropdown list of commands to select and it devotes a decent amount of space to showing me the syntax box for a Pandas data frame. I apologize if this is not clear. I've tried doing a screen capture but it does not show up well.

I guess what I'm saying is this: when I type the 'N' in 'Name', a dropdown shows me commands that begin with N. I also cannot get rid of the box showing the parameters in a pandas Dataframe (kind of like the top of the help file). I can get rid of that by turning off the inline suggestions, but then I use the autofill of getting a dataframe when I type "pd.D". I wish there was some sort of in between setting.

**Comment by isabelizimm** [comment]
Hi there 👋 Thanks for trying out Positron! I want to make sure I am understanding your experience correctly. In your example, this is what I would expect to see when trying to autocomplete a column name in a data frame. I'm demonstrating in the console, but the experience should be the same in a Python script or Jupyter Notebook.

Using `df['Name']` column selection, my cursor is between the quotes
![Screenshot 2024-09-11 at 12 34 02 PM](https://github.com/user-attachments/assets/d5e6a427-787c-4825-9f37-5d7e729c8eb2)

Using `df.Name` column selection, my cursor is at the end of the line
![Screenshot 2024-09-11 at 12 35 47 PM](https://github.com/user-attachments/assets/c3d2bd6b-4ace-40b7-8927-a73006c7c479)

Is this what you are seeing? It sounds like maybe not, and that other information + boxes are populating. If not, can you provide the version of Positron and Python you're using?

**Reply by adrennhoff** [reply]
Thanks for your response. I was able to successfully use Print Screen to capture what I was saying. I will post the images below. For reference, I am using Python 3.11.5 and Positron 2024.09.0 (build 1). The inline suggestions for you are what I want but it seems to be inferring things when I create the data frame. When I start typing "Name" as the first column, that's where the suggestions/options first appear. You can't really tell from the image but I just typed the parentheses and "N" to type "Name" and you can see that it is offering me a dropdown list of suggestions.

![image](https://github.com/user-attachments/assets/3bfc8e96-1cf0-4146-9c27-993a542a040f)

In the next image, I have created 'Name' and now I am inside the brackets attempting to type the first name ("Bob").

![image](https://github.com/user-attachments/assets/f1c29257-f867-4bdf-aa55-5de39ca9e69a)

**Reply by isabelizimm** [reply]
Oh gosh, that is not ideal at all 😨 I am able to replicate this behavior locally-- thank you so much for the screenshots! I'm going to promote this to an issue so we can find out a way to make this a nicer experience. 

EDIT: closing this discussion in favor of https://github.com/posit-dev/positron/issues/4644 to track progress 👍 The bug report is much appreciated!
